tests/test_launcher_remote.py

```diff
@@ -0,0 +1,64 @@
+"""Test the PyPIM integration."""
+from unittest.mock import create_autospec
+
+import ansys.platform.instancemanagement as pypim
+import grpc
+
+from ansys.mapdl.core.launcher import launch_mapdl
+from ansys.mapdl.core.mapdl_grpc import MAX_MESSAGE_LENGTH
+
+
+def test_launch_remote_instance(monkeypatch, mapdl):
+    # Create a mock pypim pretenting it is configured and returning a channel to an already running mapdl
+    mock_instance = pypim.Instance(
+        definition_name="definitions/fake-mapdl",
+        name="instances/fake-mapdl",
+        ready=True,
+        status_message=None,
+        services={"grpc": pypim.Service(uri=mapdl._channel_str, headers={})},
+    )
+    pim_channel = grpc.insecure_channel(
+        mapdl._channel_str,
+        options=[
+            ("grpc.max_receive_message_length", MAX_MESSAGE_LENGTH),
+        ],
+    )
+    mock_instance.wait_for_ready = create_autospec(mock_instance.wait_for_ready)
+    mock_instance.build_grpc_channel = create_autospec(
+        mock_instance.build_grpc_channel, return_value=pim_channel
+    )
+    mock_instance.delete = create_autospec(mock_instance.delete)
+
+    mock_client = pypim.Client(channel=grpc.insecure_channel("localhost:12345"))
+    mock_client.create_instance = create_autospec(
+        mock_client.create_instance, return_value=mock_instance
+    )
+
+    mock_connect = create_autospec(pypim.connect, return_value=mock_client)
+    mock_is_configured = create_autospec(pypim.is_configured, return_value=True)
+    monkeypatch.setattr(pypim, "connect", mock_connect)
+    monkeypatch.setattr(pypim, "is_configured", mock_is_configured)
```

Comment:
You might want to set this as a fixture for future tests

Reply:
@germa89 I don't think I can make this code reusable: each mocked part created in this block has assertions at the end of the test. This is pretty much the main downside of the mock approach in my opinion: the tests have a significant boilerplate.

Reply:
Fair enough.